openshift-client/src/main/java/io/fabric8/openshift/client/OpenShiftConfig.java

```diff
@@ -55,6 +58,14 @@ public OpenShiftConfig(Config kubernetesConfig) {
     );
   }
 
+  public OpenShiftConfig(Config kubernetesConfig, String apiGroupUrl, String apiGroupVersion) {
+    this(kubernetesConfig,
+      getDefaultOpenShiftUrl(kubernetesConfig), getDefaultOapiVersion(kubernetesConfig), DEFAULT_BUILD_TIMEOUT
+    );
+    this.openShiftUrl = apiGroupUrl;
```

Comment:
OK I've moved the apiGroup / apiGroupVersion state out of the OpenShiftConfig class and into the OpenShiftOperation now with a little helper class `ConfigAndApiGroupsInfo` used to minimise disruption to all the constructors of operations & handlers